File: clients/javascript/src/models/authTokenRotateIn.ts
// this file is @generated

export interface AuthTokenRotateIn {
    namespace?: string | null;
    id: string;
    prefix?: string;
    suffix?: string | null;
    /** Milliseconds from now until the old token expires. `None` means expire immediately. */
    expiryMs?: number | null;
}

export const AuthTokenRotateInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): AuthTokenRotateIn {
        return {
            namespace: object['namespace'],
            id: object['id'],
            prefix: object['prefix'],
            suffix: object['suffix'],
            expiryMs: object['expiry_ms'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: AuthTokenRotateIn): any {
        return {
            'namespace': self.namespace,
            'id': self.id,
            'prefix': self.prefix,
            'suffix': self.suffix,
            'expiry_ms': self.expiryMs,
        };
    }
}

File: clients/python/diom/models/auth_token_rotate_in.py
# this file is @generated
import typing as t
from pydantic import Field

from ..internal.base_model import BaseModel


class AuthTokenRotateIn(BaseModel):
    namespace: t.Optional[str] = None

    id: str

    prefix: t.Optional[str] = None

    suffix: t.Optional[str] = None

    expiry_millis: t.Optional[int] = Field(default=None, alias="expiry_millis")
    """Milliseconds from now until the old token expires. `None` means expire immediately."""
